src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetTaskByIdControllerTest.java

```diff
@@ -577,14 +577,14 @@ void should_return_404_when_initiation_request_failed_to_retrieve_data_from_ccd(
                 .content(asJsonString(initiateTaskRequest)))
             //.andDo(print())
             .andExpectAll(
-                status().is(HttpStatus.INTERNAL_SERVER_ERROR.value()),
+                status().isBadGateway(),
                 content().contentType(APPLICATION_PROBLEM_JSON_VALUE),
-                jsonPath("$.type")
-                    .value("https://github.com/hmcts/wa-task-management-api/problem/generic-server-error"),
-                jsonPath("$.title").value("Generic Server Error"),
-                jsonPath("$.status").value(500),
-                jsonPath("$.detail").value("Generic Server Error: The action could not be "
-                                           + "completed because there was a problem when initiating the task.")
+                jsonPath("$.type").value("https://github.com/hmcts/wa-task-management-api/problem/downstream-dependency-error"),
+                jsonPath("$.title").value("Downstream Dependency Error"),
+                jsonPath("$.status").value(502),
+                jsonPath("$.detail").value(
+                    "Downstream dependency did not respond as expected and the request could not be completed."
+                        + " Message from downstream system: [401 Unauthorized] during [GET] to [http://ccd-data-store-api-aat.service.core-compute-aat.internal/cases/getTaskCaseId3] [CcdDataServiceApi#getCase(String,String,String)]: []")
```

Comment:
Maybe I'm missing something.  Since this is an integration test where is this message coming from?  It seems very specifically a 401 when above on line 554 you're returning a generic RuntimeException with a different message from CCD?

Reply:
FiegnClient is wrapping the RuntimeException with Feign Exception Ravi. I wondered the same thing regarding the message. :) Should I investigate more?

Reply:
Yes please ... maybe the when clause is missing

Reply:
Hi Ravi, The message is part of FeignException. It used to throw GenericException, but now with these new changes we are re-throwing feign exception. By default its setting the message as displayed.

Reply:
Hi Ravi... I found the problem. Instead of using @MockBean someone use @Mock annotation due to that ccdDataServiceApi is loaded without the spring context and the Mock is also not working as expected. I changed it to MockBean annotation. Now its working as expected and also all the tests are passing in local.